### ISSUE TITLE
Egui WASM program failing or not failing depending on whitespace/comments and other irrelevant details

### ISSUE THREAD

**Issue by hkBst** [rust-lang/rust#119774]
Egui WASM program failing or not failing depending very sensitively on whitespace/comments and other irrelevant details, similar to https://github.com/rust-lang/rust/issues/110230 but without the ICE.

See https://github.com/emilk/egui/issues/3796#issuecomment-1883097612 for description and test case. 

`rustc --version --verbose`:
```
rustc 1.74.1 (a28077b28 2023-12-04) (gentoo)
binary: rustc
commit-hash: a28077b28a02b92985b3a3faecf92813155f1ea1
commit-date: 2023-12-04
host: x86_64-unknown-linux-gnu
release: 1.74.1
LLVM version: 17.0.6
<version>
```

**Comment by jieyouxu** [comment]
> Actually this problem seems to be more fundamental and is possibly a compiler bug. I've now included below a minimal test case, which still works (for me), but which stops working when I remove comments or add blank lines or change button strings, or remove the game: Game, member from App.

Do you have any error outputs from rustc? What do you mean by "stops working"? Are there any Internal Compiler Errors (ICEs) shown?

**Comment by hkBst** [comment]
Rustc does not give any error output, but acts like all is fine. Still the created web application does not work and shows an empty screen in the browser.
